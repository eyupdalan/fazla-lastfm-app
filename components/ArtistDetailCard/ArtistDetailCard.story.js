import ArtistDetailCard from "./index";

const artistDetailCardStory = {
    title: 'Components/ArtistDetailCard',
    component: ArtistDetailCard,
};

export const ArtistDetailCardStory = () => (
    <ArtistDetailCard
        artist={"test"}
        name={"test"}
        image={"https://upload.wikimedia.org/wikipedia/commons/4/42/LinkinParkBerlin2010.jpg"}
        mbid={"test"}
        playCount={1000}
        listeners={2000}
    />
);
export const ArtistDetailCardStoryWithoutListeners = () => (
    <ArtistDetailCard
        artist={"test"}
        name={"test"}
        image={"https://upload.wikimedia.org/wikipedia/commons/4/42/LinkinParkBerlin2010.jpg"}
        mbid={"test"}
        playCount={1000}
    />
);

export default artistDetailCardStory;