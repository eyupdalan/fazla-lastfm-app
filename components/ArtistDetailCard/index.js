import styles from "../../styles/ArtistDetailCard.module.css";
import Image from "../Image";
import ArtistName from "../ArtistName";
import PropTypes from "prop-types";
import {Card} from "antd";

export default function ArtistDetailCard({mbid, artist, name, image, listeners, playCount}) {
    return (
        <Card style={{width: 500, margin: 10}}>
            <div className={styles.artistDetailCard}>
                <div className={styles.artistDetailCardImage}>
                    <Image src={image} alt={name} height={100} width={100}/>
                </div>
                <div className={styles.artistDetailCardContent}>
                    <div>
                        <div>{name}</div>
                        <ArtistName mbid={mbid} name={artist}/>
                    </div>
                    <div>
                        {listeners && <div>Listeners: {listeners}</div>}
                        <div>Play count: {playCount}</div>
                    </div>
                </div>
            </div>
        </Card>
    )
}

ArtistDetailCard.propTypes = {
    mbid: PropTypes.string.isRequired,
    artist: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
    listeners: PropTypes.number,
    playCount: PropTypes.number
}

ArtistDetailCard.defaultProps = {
    listeners: null,
    playCount: null
}