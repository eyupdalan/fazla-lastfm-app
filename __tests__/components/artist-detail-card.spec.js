import { render } from '@testing-library/react'
import ArtistDetailCard from '../../components/ArtistDetailCard';

it('renders Artist Detail Card component', () => {
    const { container } = render(<ArtistDetailCard  artist={"test"} name={"test"} mbid={"test"} image={"https://upload.wikimedia.org/wikipedia/commons/4/42/LinkinParkBerlin2010.jpg"} playCount={5000} listeners={6000} />)
    expect(container).toMatchSnapshot()
})
