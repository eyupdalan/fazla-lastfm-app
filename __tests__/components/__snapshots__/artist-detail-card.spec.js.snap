// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Artist Detail Card component 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered"
    style="width: 500px; margin: 10px;"
  >
    <div
      class="ant-card-body"
    >
      <div
        class="artistDetailCard"
      >
        <div
          class="artistDetailCardImage"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27100%27%20height=%27100%27/%3e"
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              alt="test"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
        </div>
        <div
          class="artistDetailCardContent"
        >
          <div>
            <div>
              test
            </div>
            <div
              class="artistName"
            >
              <a
                href="/artist/test"
              >
                test
              </a>
            </div>
          </div>
          <div>
            <div>
              Listeners: 
              6000
            </div>
            <div>
              Play count: 
              5000
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
